clusterloader2/api/validation/validation.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/perf-tests/clusterloader2/api"
+)
+
+// VerifyConfig does simple validation for config and all included structures.
+// Returns error is any condition is not satisfied.
+func VerifyConfig(c *api.Config) field.ErrorList {
+	allErrs := field.ErrorList{}
+	fldPath := field.NewPath(".")
+	if c.AutomanagedNamespaces < 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath, "AutomanagedNamespaces", "must be zero or more"))
+	}
+	for i, tuningSet := range c.TuningSets {
+		allErrs = append(allErrs, verifyTuningSet(&tuningSet, fldPath.Child("TuningSets").Index(i))...)
```

Comment:
all "Child" methods reference to json representation of a field. So it should start with small letter (it.s json tag value)